### DISCUSSION TITLE
Use SSR and a CDN like cloudfront

### DISCUSSION THREAD

**Discussion by woile** [nuxt/nuxt#9408] [Questions (Nuxt 2)] [answered]
Hello people, I'd like to know how can this be done and anyone experience.
I would like to have the SSR functionality, while also serving the statics from a cdn like cloudfront.

The way I understand it, the first hit from the browser, let's say `/`, should be handled by my nuxt app (running on kubernetes, EC2, or whatever). But once the browser starts processing the HTML, the static files, like the `js`, `css` or images should be lifted from a CDN.
The examples for cloudfront on aws only make mention of `static` deployment.
https://nuxtjs.org/docs/2.x/deployment/deployment-amazon-web-services

Thanks in advance

**Comment by btkostner** [comment, marked as answer]
So, first off, CloudFormation is not a CDN, you are probably thinking about CloudFront (AWS naming is super confusing.)

There are a couple ways of doing this. If you want to use CloudFront you can set the origin to the server serving your SSR content. CloudFront will read and cache your statically built assets. Then just access your site with the CloudFront domain. [More information](https://aws.amazon.com/blogs/aws/amazon-cloudfront-support-for-custom-origins/). One thing to note here is that CloudFront has a 30 second response time limit, so if you have pages that would take longer than 30 seconds to respond, you will need to do something else.

Another way to do it use CloudFront on an alternative domain. You'd still setup the origin server, but instead of sending all traffic through CloudFront, you will only use it for the static files. `example.com` would be where users would view your site, but you'd also have `assets.example.com` (or something similar.) Then just change your Nuxt config to serve built files from that domain.

And lastly, you can try some other CDN services like Cloudflare. Cloudflare is free, and you just need to setup the DNS for your server.

**Comment by woile** [comment]
Thank you so much! I'm gonna give them a try. I think I would go for the second case.
I corrected the name for search engines.

Just for clarification where is the built files config?

I think is this one, right?
https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-build#publicpath

**Reply by btkostner** [reply]
Yes, that is correct. You'll want it to point to the cloudfront domain, so you'll end up setting the `publicPath` to something like `https://d111111abcdef8.cloudfront.net/_nuxt/` (by default.) You can also change that domain if you want it to be a subdomain instead.
